src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs

```diff
@@ -110,6 +110,12 @@ private void setSecondaryColumnUnit(IReadOnlyList<ColumnInfo> columnInfos)
 
       protected abstract UnitDescription ExtractUnits(string description, IUnformattedData data, List<string> keys, IReadOnlyList<IDimension> supportedDimensions,  ref double rank);
 
+      public UnitDescription ExtractUnits(string description, IReadOnlyList<IDimension> supportedDimensions)
+      {
+         var rank = 0.0;
```

Comment:
what is this rank again? and should not the rank be an int?

Reply:
rank is a number that will indicate how good a match is a format for a given file. So, if there are two formats (right now we have only one but we discussed about having more at some point) that are valid for a file, the rank parameter will tell how much of a format specific feature was detected so the format with the higher rank should be selected. If you have a file with units on the headers, it should also be possible to use the format with the units on the columns but the one with units on the headers should be the first option since the file has this specific feature. I hope it makes more sense now but we could off course discuss it.

Reply:
I see. Not sure why rank is double and not an integer but this works